src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyCacheMaxSizeAction.java

```diff
@@ -109,12 +124,11 @@ private void setBounds() {
         // This is intentionally not made static because different nodes can
         // have different bounds based on instance types
 
-        // TODO: Read the upperBound from NodeConfigurationRca.
         // Field data cache used when sorting on or computing aggregation on the field (in Bytes)
-        upperBoundInBytes.put(ResourceEnum.FIELD_DATA_CACHE, 12000 * 1_000_000L);
+        upperBoundInBytes.put(ResourceEnum.FIELD_DATA_CACHE, (long) (heapMaxSizeInBytes * fieldDataCacheSizeUpperBound));
```

Comment:
do we need any assertion for the `heapMaxSizeInBytes` value ? Maybe non-null, non-zero ?

Reply:
Decider checks for non-null, I will add a check for non-zero in the decider.